mocha-client-tests/karma.conf.js

```diff
@@ -0,0 +1,72 @@
+module.exports = function(config) {
+  config.set({
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '',
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['mocha'],
+
+    // list of files / patterns to load in the browser
+    files: [
+      // Test dependencies
+      'node_modules/expect.js/index.js',
+      'node_modules/sinon-chrome/bundle/sinon-chrome-webextensions.min.js',
```

Comment:
@sneakypete81 when in the WebExtensions API schema files there is a `"async"` property on a method definition with a callback parameter, the callback parameter can be omitted and that API method will return a Promise, e.g. for the `StorageArea.get` is at line https://hg.mozilla.org/mozilla-central/file/tip/toolkit/components/extensions/schemas/storage.json#l37

(the value of the `"async"` property is the name of the callback parameter, that is almost always `"callback"`)